numpy/lib/arraysetops.py

```diff
@@ -239,59 +295,124 @@ def reshape_uniq(uniq):
         uniq = np.swapaxes(uniq, 0, axis)
         return uniq
 
+    orig_base = consolidated
+    while (orig_base.base is not None and orig_base is not orig_ar
+           and orig_base.base is not orig_ar):
+        orig_base = orig_base.base
```

Comment:
What exactly are you trying to do here? I think this deserves some comments, and the `orig_base is orig_ar or orig_base.base is orig_ar` code from below calculated alongside it.

Reply:
Or better, a helper function

Reply:
Trying to figure out whether `consolidated` is a view whose underlying data points to `orig_ar` even indirectly. I found no helper function for that. I could add one I guess.

Reply:
Done, added `np.array_view_root` with corresponding tests.